### PR TITLE
FIX-1998: Split RNN Cell Module

### LINKED ISSUE #1998: Split RNN Cell Module
Our RNN Cell module contains several different implementations and it makes it difficult for us to correctly assign owners to each cell:
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/cell.py
https://github.com/tensorflow/addons/blob/master/.github/CODEOWNERS#L166-L167

We should instead split them into their own modules along with their test cases.

### PR DESCRIPTION
**Pull request by marload**
# Description

Please include a summary of the change and which issue is fixed. 

Fixes #1998 

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition (Requires creating an issue first)
    - [ ] The changes conform to the [contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*  

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [x] By running pre-commit hooks
- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

### CODE REVIEW

**Comment by marload** [comment]
@qlzh727 

Thank you. I have updated CODEOWNER. Perhaps the reason why the CODEOWNER test failed is because the existing file was deleted and the owner was changed. I don't know how to solve this error.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -163,8 +163,16 @@
 /tensorflow_addons/optimizers/yogi.py @manzilz
 /tensorflow_addons/optimizers/tests/yogi_test.py @manzilz
 
-/tensorflow_addons/rnn/cell.py @qlzh727 @pedrolarben
-/tensorflow_addons/rnn/tests/cell_test.py @qlzh727 @pedrolarben
+/tensorflow_addons/rnn/esn_cell.py @qlzh727 @pedrolarben
+/tensorflow_addons/rnn/tests/esn_cell_test.py @qlzh727 @pedrolarben
+/tensorflow_addons/rnn/layer_norm_lstm_cell.py @qlzh727 @pedrolarben
+/tensorflow_addons/rnn/tests/layer_norm_lstm_cell/cell_test.py @qlzh727 @pedrolarben
```
Comment (WindQAQ) [inline .github/CODEOWNERS:169]: I believe it's because you make a typo here
```
/tensorflow_addons/rnn/tests/layer_norm_lstm_cell_test.py
```

**Comment by marload** [comment]
@qlzh727 
Can you tell me why "GitHub CI / Check that we can build the docs" failed?

**Comment by seanpmorgan** [comment]
> @qlzh727
> Can you tell me why "GitHub CI / Check that we can build the docs" failed?

It's likely unrelated to this PR and because of https://github.com/tensorflow/addons/issues/2043

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -163,8 +163,16 @@
 /tensorflow_addons/optimizers/yogi.py @manzilz
 /tensorflow_addons/optimizers/tests/yogi_test.py @manzilz
 
-/tensorflow_addons/rnn/cell.py @qlzh727 @pedrolarben
-/tensorflow_addons/rnn/tests/cell_test.py @qlzh727 @pedrolarben
+/tensorflow_addons/rnn/esn_cell.py @qlzh727 @pedrolarben
```
Comment (qlzh727) [inline .github/CODEOWNERS:166]: The order here is quite massy, can u sort them?

Comment (qlzh727) [inline .github/CODEOWNERS:166]: For esn_cell and its test, you can put the owner to @pedrolarben, and the rest of them, you can leave them to me.

Comment (marload) [reply .github/CODEOWNERS]: applied 😄

**Pull request merged**